### ISSUE TITLE
comment argument 

### ISSUE THREAD

**Issue by orhanabar** [JuliaData/CSV.jl#361]
Comment works for other rows but the first row.
If first row after the header is starting with the comment string, it still parses it.

**Comment by orhanabar** [comment]
But as you can see here 2nd and 3rd rows does not show up. However, first row is still there.

![image](https://user-images.githubusercontent.com/43005621/49324648-18391780-f500-11e8-88cb-bd276ca07d1f.png)

**Comment by quinnj** [comment]
Seems to be working for me:
```julia
julia> df = CSV.File(joinpath(dirname(pathof(CSV)), "../test/testfiles/test_comment_361.csv"), comment="#") |> DataFrame
2×6 DataFrame
│ Row │ Name     │ Term   │ Prerequisites │ Corequisites │ Credits │ Passrate │
│     │ String⍰  │ Int64⍰ │ Missing       │ Missing      │ Int64⍰  │ Missing  │
├─────┼──────────┼────────┼───────────────┼──────────────┼─────────┼──────────┤
│ 1   │ MATH 125 │ 1      │ missing       │ missing      │ 4       │ missing  │
│ 2   │ ECE 121  │ 1      │ missing       │ missing      │ 1       │ missing  │
```

**Comment by orhanabar** [comment]
That's wired. I sent the file and output. Are you sure that you are using the same version v0.4.2?

**Comment by quinnj** [comment]
I'm on current master; where did you send the file?

**Comment by orhanabar** [comment]
I didn't understand your question.
By the way, I tried to remove and install CSV. However, the problem remains.

**Comment by quinnj** [comment]
Can you upload the exact file here so I can try the exact same file on my machine?

**Comment by orhanabar** [comment]
Here is the file 
[Alabama EESheet1.zip](https://github.com/JuliaData/CSV.jl/files/2635646/Alabama.EESheet1.zip)

**Comment by orhanabar** [comment]
Other than that issue, shouldn't it also work if string is in double quotes like:
```
Name,Term,Prerequisites,Corequisites,Credits,Passrate
"#CH 101,1",,"MATH 125",4,
"#MATH 125",1,,,4,
"#ECE 121",1,,,1,
"EN 101",1,,,3,
```
When I run for this lines it shows all lines.

**Comment by quinnj** [comment]
No, if the comment character/string isn't the very first characters of a row, they're not considered "comments" and the row won't be ignored.

**Comment by quinnj** [comment]
Yeah, I don't have any problems with the exact file you shared:
```julia
julia> df = CSV.read(file, comment="#")
39×6 DataFrame
│ Row │ Name                   │ Term   │ Prerequisites    │ Corequisites       │ Credits │ Passrate │
│     │ Union{Missing, String} │ Int64⍰ │ String⍰          │ String⍰            │ Int64⍰  │ Missing  │
├─────┼────────────────────────┼────────┼──────────────────┼────────────────────┼─────────┼──────────┤
│ 1   │ MATH 125               │ 1      │ missing          │ missing            │ 4       │ missing  │
│ 2   │ ECE 121                │ 1      │ missing          │ missing            │ 1       │ missing  │
│ 3   │ EN 101                 │ 1      │ missing          │ missing            │ 3       │ missing  │
│ 4   │ HU/L/FA                │ 1      │ missing          │ missing            │ 3       │ missing  │
│ 5   │ PH 105                 │ 2      │ MATH 125         │ missing            │ 4       │ missing  │
│ 6   │ MATH 126               │ 2      │ MATH 125         │ missing            │ 4       │ missing  │
│ 7   │ ENGR 103               │ 2      │ MATH 125         │ missing            │ 3       │ missing  │
│ 8   │ EN 102                 │ 2      │ EN 101           │ missing            │ 3       │ missing  │
```

I think something must be getting messed up in your versions somehow.

**Comment by quinnj** [comment]
Even with 0.4.2, I see the same thing. Working. Can you show the exact output of `using Pkg; Pkg.installed()` to confirm your package versions?

**Comment by orhanabar** [comment]
```
julia> using Pkg; Pkg.installed()
Dict{String,Union{Nothing, VersionNumber}} with 31 entries:
  "CSV"                   => v"0.4.2"
  "Statistics"            => nothing
  "GLPK"                  => v"0.8.3"
  "LightGraphs"           => v"1.2.0"
  "SQLite"                => v"0.7.0"
  "Clp"                   => v"0.5.1"
```


```
(v1.0) pkg> st
    Status `C:\Users\orhan\.julia\environments\v1.0\Project.toml`
  [ad839575] Blink v0.8.1
  [a076750e] CPLEX v0.4.1
  [336ed68f] CSV v0.4.2
  [9961bab8] Cbc v0.4.2
  [e2554f3b] Clp v0.5.1
```

**Comment by orhanabar** [comment]
> Works here with CSV 0.4.3.
@nalimilan  I tried but it keeps adding version 0.4.2. How did you install 0.4.3?

**Comment by nalimilan** [comment]
Then there's probably something in your installation which blocks the upgrade, and which might indicate a deeper problem. Can you try starting Julia with the `JULIA_DEPOT_PATH` environment variable pointing to an empty directory?

**Comment by quinnj** [comment]
@orhanabar , you get this figured out? I'm closing the issue for now, but we can still try to help w/ any Pkg-releated issues you might be having.

**Comment by orhanabar** [comment]
I have created a function to remove comment line because it should consider a line as comment line if it starts with "# too. Asking user not to rely on csv editor and use a text editor to make sure comment charecter is the very first character is not a good way of doing it for me (it doesn't accept comment="#|\"#" either). That's why I didn't try it and commented here. 

Today I tried and it is 0.4.3 version is working the way you expected.
